crates/ruff/src/rules/pylint/rules/import_outside_top_level.rs

```diff
@@ -0,0 +1,44 @@
+use rustpython_parser::ast::Stmt;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::prelude::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `import` statements that are not at the top-level of a file.
+///
+/// ## Why is this bad?
+/// According to [Pep 8 - Style Guide for Python Code](https://peps.python.org/pep-0008/#imports),
+/// > Imports are always put at the top of the file, just after any module comments and docstrings,
+/// and before module globals and constants.
+///
+/// ## Example
+/// ```python
+/// def foo():
+///     import os
+///
+///     ...
+/// ```
+///
+/// ## References
+/// - [PEP 8 - Style Guide for Python Code](https://peps.python.org/pep-0008/#imports)
+#[violation]
+pub struct ImportOutsideTopLevel;
+
+impl Violation for ImportOutsideTopLevel {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`import` should be used only at the top level of a file")
+    }
+}
+
+/// C0415
+pub(crate) fn import_outside_top_level(checker: &mut Checker, stmt: &Stmt) {
+    if !checker.semantic().at_top_level() && !checker.semantic().in_type_checking_block() {
```

Comment:
As far as I can tell, Pylint only flags this for imports outside of the module scope:

```python
if True:
  import sys  # Pylint doesn't flag this.

def foo():
  import sys  # It does flag this!
```

I think this should perhaps be `!checker.scope().kind.is_module()`?